src/components/structures/ContextMenu.tsx

```diff
@@ -26,6 +26,7 @@ import { Writeable } from "../../@types/common";
 import { replaceableComponent } from "../../utils/replaceableComponent";
 import UIStore from "../../stores/UIStore";
```

Comment:
> ❌ `RoomSubList` (the settings `...` on each room list heading for sorting stuff)
> * Something is very strange with this menu. On `develop`, it requires 2 clicks to change the radios and you can't navigate to the checkboxes. Also when using arrow keys on the radios, it seems to be the opposite of the focused selection.
> * In this PR, I can't even get the radios in the menu to work with my mouse. It seems to only re-render when I check/uncheck the checkboxes below. Any tips?
>
> -- https://github.com/matrix-org/matrix-react-sdk/pull/7353/files#r768487459

This is the last `ContextMenu` that is being weird. It might have to do with the nested `RoamingTabIndex` in that area although the `RoomListHeader` one works just fine.

Reply:
Actually testing on the latest `develop`, I am seeing the exact same problems as I do on this branch.

> Something is very strange with this menu. On develop, it requires 2 clicks to change the radios and you can't navigate to the checkboxes. Also when using arrow keys on the radios, it seems to be the opposite of the focused selection.

This was actually tested on `develop.element.io` on Windows which sorta worked.

Reply:
> I can't even get the radios in the menu to work with my mouse. It seems to only re-render when I check/uncheck the checkboxes below. Any tips?


This can be maximally reproduced every time for me by collapsing the room list so nothing changes in the room list. Then try changing the radios, nothing will update.

I can get things to work correctly, once I expand the room list and add `aaa` and `zzz` rooms which actually re-render differently when I change sort.

Previously, I only had a few rooms which were sorted correctly under both settings so I was able to get the same effect as collapsing the room list.

Reply:
It looks like we're missing a `forceUpdate` in `onTagSortChanged`

Reply:
> radios and you can't navigate to the checkboxes

This looks like it's because radios have a native behavior to consume arrow keys and rollover to the top of the list once you reach the bottom.

I think we're going to have to allow `Tab` when you're in the radio section to jump to the next thing which is how you would interact in a normal form.

Reply:
Updated to fix `RoomSubList` and generalized form element interaction within `RovingTabIndex` and `ContextMenu` so we don't need special cases for the `ContextMenu` in `UserMenu` for example.